sway/tree/view.c

```diff
@@ -944,14 +944,16 @@ bool view_is_visible(struct sway_view *view) {
 			&& floater->parent->parent->type != C_WORKSPACE) {
 		floater = floater->parent;
 	}
-	bool is_sticky = container_is_floating(floater) && floater->is_sticky;
+	bool is_floating = container_is_floating(floater);
+	bool is_sticky = is_floating && floater->is_sticky;
 	// Check view isn't in a tabbed or stacked container on an inactive tab
 	struct sway_seat *seat = input_manager_current_seat(input_manager);
 	struct sway_container *container = view->swayc;
 	while (container->type != C_WORKSPACE) {
 		if (container->parent->layout == L_TABBED ||
 				container->parent->layout == L_STACKED) {
-			if (seat_get_active_child(seat, container->parent) != container) {
+			if (!is_floating && seat_get_active_child(seat,
```

Comment:
If you have a floating tabbed container, and the view being checked is the active child of the tabbed container, this will return false which is incorrect. `is_floating` will be true for any child of the floating container.

I think if you change the `while` condition to be `container->type != C_WORKSPACE && container->layout != L_FLOATING` then that would solve it.

Reply:
And then we'll read `workspace->sway_workspace` after the loop :\

Instead I checked if `container` is floating. If a floating container has a direct tabbed/stacked parent, it's always visible. Does that sound better to you?

Reply:
My bad, you're right. Force pushed your solution.